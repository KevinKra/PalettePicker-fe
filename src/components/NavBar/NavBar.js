import React from 'react';
import './NavBar.scss';

export default function NavBar() {
	return (
		<header className="NavBar">
			<h2>PalettePicker</h2>
			<form className="navbar-form">
				<button>Login</button>
				<button>Register</button>
			</form>
		</header>
	);
}
