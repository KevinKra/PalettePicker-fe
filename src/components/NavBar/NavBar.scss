@import "../../styles/main.scss";

.NavBar {
	position: sticky;
	top: 0;
	width: 100%;
	display: flex;
	align-items: center;
	justify-content: space-between;
	z-index: 5;
	background-color: $color-primary;
	border-bottom: 2px solid $color-accent;
	box-shadow: -1px -5px 5px 6px rgba(0, 0, 0, 0.75);

	h2 {
		margin: 0.5rem;
		margin-left: 3.5rem;
		font-family: "Carter One", cursive;
		color: coral;
	}
	.navbar-form {
		display: flex;
		margin-right: 3.5rem;
	}
	.navbar-form > button {
		@include btn-style-1;
		padding: 0.4rem 2.5rem;
		margin: 0.2rem;
	}
}
