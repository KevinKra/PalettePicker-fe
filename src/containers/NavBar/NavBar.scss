.NavBar {
  position: sticky;
  top: 0;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  z-index: 5;
  background-color: white;
  border-bottom: 2px solid coral;
  box-shadow: -1px -5px 5px 6px rgba(0, 0, 0, 0.75);
  h2 {
    margin: 0.5rem;
    margin-left: 3.5rem;
    font-family: "Carter One", cursive;
    color: coral;
  }

  .navbar-form {
    margin-right: 3.5rem;
  }

  .navbar-form > button {
    padding: 0.4rem 2.5rem;
    margin: 0.2rem;
    color: white;
    background-color: coral;
    border: 1px solid rgb(199, 94, 56);
    border-radius: 5px;
    cursor: pointer;
    font-size: 14px;
  }
}
