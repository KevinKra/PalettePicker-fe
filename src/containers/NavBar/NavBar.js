import React, { Fragment } from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import * as actions from "../../_redux/actions";
import "./NavBar.scss";

export const NavBar = props => {
  const handleClick = (e, destination) => {
    e.preventDefault();
    props.history.push(`/${destination || ""}`);
  };

  const loginFormat =
    Object.keys(props.userDetails).length === 0 ? (
      <Fragment>
        <button data-test="login" onClick={e => handleClick(e, "login")}>
          Login
        </button>
        <button data-test="register" onClick={e => handleClick(e, "register")}>
          Register
        </button>
      </Fragment>
    ) : (
      <button data-test="logout" onClick={e => handleLogout(e)}>
        Logout
      </button>
    );

  const handleLogout = e => {
    e.preventDefault();
    props.logoutUser();
    handleClick(e);
  };

  return (
    <header className="NavBar">
      <h2>PalettePicker</h2>
      <form className="navbar-form">{loginFormat}</form>
    </header>
  );
};

const mapStateToProps = store => ({
  userDetails: store.userDetails
});

const mapDispatchToProps = dispatch => ({
  logoutUser: () => dispatch(actions.logoutUser())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(NavBar));
