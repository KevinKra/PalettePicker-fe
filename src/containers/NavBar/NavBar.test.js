import React from "react";
import { NavBar } from "./NavBar";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("NavBar", () => {
  let wrapper;
  let mockUserDetails = { username: "steve", id: 123 };
  const mockHandleLogout = jest.fn();
  const historyMock = { push: jest.fn() };
  beforeEach(() => {
    wrapper = shallow(
      <NavBar
        userDetails={mockUserDetails}
        logoutUser={jest.fn()}
        handleLogout={mockHandleLogout}
        history={historyMock}
      />
    );
  });
  it("should match the snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("handleClick method should push to the '/' path if no argument is provided (logout)", () => {
    const button = wrapper.find('[data-test="logout"]');
    button.simulate("click", { preventDefault() {} });
    expect(historyMock.push.mock.calls[0]).toEqual(["/"]);
  });
  it("handleClick: should update the url path to login if login passed as argument", () => {
    const wrapper = shallow(<NavBar userDetails={{}} history={historyMock} />);
    const button = wrapper.find('[data-test="login"]');
    button.simulate("click", {
      preventDefault: () => {},
      destination: "login"
    });
    expect(historyMock.push.mock.calls[1]).toEqual(["/login"]);
  });
});
