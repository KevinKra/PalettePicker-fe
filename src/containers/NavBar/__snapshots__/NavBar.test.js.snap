// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the snapshot 1`] = `
<header
  className="NavBar"
>
  <h2>
    PalettePicker
  </h2>
  <form
    className="navbar-form"
  >
    <button
      data-test="logout"
      onClick={[Function]}
    >
      Logout
    </button>
  </form>
</header>
`;
